libsolidity/analysis/ControlFlowAnalyzer.cpp

```diff
@@ -145,3 +146,51 @@ void ControlFlowAnalyzer::checkUninitializedAccess(CFGNode const* _entry, CFGNod
 		}
 	}
 }
+
+void ControlFlowAnalyzer::checkUnreachable(CFGNode const* _entry, CFGNode const* _exit, CFGNode const* _revert) const
+{
+	std::set<CFGNode const*> reachable;
+	std::set<CFGNode const*> nodesToTraverse;
+
+	// collect all nodes reachable from the entry point
+	nodesToTraverse.insert(_entry);
+	while (!nodesToTraverse.empty())
+	{
+		CFGNode const* node = *nodesToTraverse.begin();
+		nodesToTraverse.erase(nodesToTraverse.begin());
+
+		if (reachable.insert(node).second)
+			for (CFGNode const* exit: node->exits)
+				nodesToTraverse.insert(exit);
+	}
+
+	// traverse all paths backwards from exit and revert
+	// and extract (valid) source locations of unreachable nodes into sorted set
+	std::set<CFGNode const*> visited;
+	std::set<SourceLocation> unreachable;
+	nodesToTraverse.insert(_exit);
+	nodesToTraverse.insert(_revert);
+	while (!nodesToTraverse.empty())
```

Comment:
This could call the same helper function

Reply:
Yes - and here I in fact *do* the check you suggested above :-).